strawberry/utils/await_maybe.py

```diff
@@ -8,7 +8,7 @@
 
 
 async def await_maybe(value: AwaitableOrValue):
-    if inspect.iscoroutine(value):
+    if inspect.isawaitable(value):
```

Comment:
Is this change related to the version bump?

Reply:
Yeah, mypy 0.910 is ok with this but mypy 0.920 gives:
`Incompatible types in "await" (actual type "CoroutineType", expected type "Awaitable[Any]")`.

We could alter the typing of the function to only accepts coroutines, instead of changing to `isawaitable`.

Reply:
would you be come sending a separate PR just for this? 😊 

I think it is worth isolating this change 😊 I also want to make sure there's no breaking change (even though our tests are passing)

Reply:
Sure, here's a separate PR - https://github.com/strawberry-graphql/strawberry/pull/1505 :)